hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/snapshot/SnapshotDiffManager.java

```diff
@@ -838,6 +835,18 @@ private synchronized void updateJobStatus(String jobKey,
     snapDiffJobTable.put(jobKey, snapshotDiffJob);
   }
 
+  private synchronized void updateJobStatusToDone(String jobKey,
+                                                  long totalNumberOfEntries) {
+    SnapshotDiffJob snapshotDiffJob = snapDiffJobTable.get(jobKey);
+    if (snapshotDiffJob.getStatus() != IN_PROGRESS) {
```

Comment:
What if its in IN_QUEUE status and the system got restarted, wouldn't you move it directly to DONE ?

Reply:
It should not matter because `updateJobStatusToDone` is called and should be called only after we successfully finish the diff calculation and report generation. If order of updating job status is messed up, we could be in that situation.

I created this function because we are updating job status along with total number of diff entries.